src/storm/mesos/MesosNimbus.java

```diff
@@ -97,57 +105,93 @@ public String getHostName(Map<String, SupervisorDetails> map, String nodeId) {
     return nodeId;
   }
 
+  @SuppressWarnings("unchecked")
   @Override
   public void prepare(Map conf, String localDir) {
     try {
-      _conf = conf;
-      _state = new LocalState(localDir);
-      String id = (String) _state.get(FRAMEWORK_ID);
-
-      _allowedHosts = listIntoSet((List) _conf.get(CONF_MESOS_ALLOWED_HOSTS));
-      _disallowedHosts = listIntoSet((List) _conf.get(CONF_MESOS_DISALLOWED_HOSTS));
-
-      _scheduler = new NimbusScheduler();
-
-      Number failoverTimeout = Optional.fromNullable((Number) conf.get(CONF_MASTER_FAILOVER_TIMEOUT_SECS)).or(3600);
-      String role = Optional.fromNullable((String) conf.get(CONF_MESOS_ROLE)).or("*");
-      Boolean checkpoint = Optional.fromNullable((Boolean) conf.get(CONF_MESOS_CHECKPOINT)).or(false);
-      String framework_name = Optional.fromNullable((String) conf.get(CONF_MESOS_FRAMEWORK_NAME)).or("Storm!!!");
-
-      FrameworkInfo.Builder finfo = FrameworkInfo.newBuilder()
-          .setName(framework_name)
-          .setFailoverTimeout(failoverTimeout.doubleValue())
-          .setUser("")
-          .setRole(role)
-          .setCheckpoint(checkpoint);
-
-      if (id != null) {
-        finfo.setId(FrameworkID.newBuilder().setValue(id).build());
-      }
-
-      Integer port = (Integer) _conf.get(CONF_MESOS_LOCAL_FILE_SERVER_PORT);
-      LOG.info("Using local port: " + port);
-      _localFileServerPort = Optional.fromNullable(port);
-
-      _httpServer = new LocalFileServer();
-      _configUrl = _httpServer.serveDir("/conf", "conf", _localFileServerPort);
-      LOG.info("Started serving config dir under " + _configUrl);
-
-      MesosSchedulerDriver driver =
-          new MesosSchedulerDriver(
-              _scheduler,
-              finfo.build(),
-              (String) conf.get(CONF_MASTER_URL));
-
-      driver.start();
-      LOG.info("Waiting for scheduler to initialize...");
-      _scheduler.waitUntilRegistered();
-      LOG.info("Scheduler initialized...");
+        initialize(conf,localDir);
+
+        MesosSchedulerDriver driver = createMesosDriver();
+        
+        driver.start();
+      
+        LOG.info("Waiting for scheduler to initialize...");
+        
+        getScheduler().waitUntilRegistered();
+        
+        LOG.info("Scheduler initialized...");
+      
     } catch (Exception e) {
       throw new RuntimeException(e);
     }
   }
-
+  
+  @SuppressWarnings("unchecked")
+  protected void initialize(Map conf, String localDir) throws Exception {
+      
```

Comment:
You do not like the @SuppressWarnings("unchecked") ??

In general I like to suppress warnings that are not going to get fixed
since the warning in that case adds nothing and its annoying from a
completeness stand point.

But I will remove it.

Thanks

On Fri, May 1, 2015 at 5:26 PM, Timothy Chen <notifications@github.com>
wrote:

> In src/storm/mesos/MesosNimbus.java
> <https://github.com/mesos/storm/pull/37#discussion_r29534976>:
>
> >      } catch (Exception e) {
> >        throw new RuntimeException(e);
> >      }
> >    }
> > -
> > +
> > +  @SuppressWarnings("unchecked")
> > +  protected void initialize(Map conf, String localDir) throws Exception {
> > +
>
> Please kill this new line
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesos/storm/pull/37/files#r29534976>.
>